### PR TITLE
MOS-685 make small improvements in various places of the code

### PR DESCRIPTION
**Pull request by awulkiew**
- Forward arguments
- Remove unused code
- Remove auto return type when it's not needed
- Add const to member functions
- Use startsWith and endsWith from utils
- Allow instantiation of operators only for specific types
- Caluclate key mask in compile time

**Your checklist for this pull request**

Make sure that this PR:
- [X] Complies with our guidelines for contributions
- [X] Has unit tests if possible

### CODE REVIEW

**Review by swichu91 — APPROVED** [review]
:)

File: module-gui/gui/Common.hpp [hunk 1]
```diff
@@ -79,60 +79,72 @@ namespace gui
...
+        enum class RectangleYap
+        {
+            None        = 0x00,
+            TopLeft     = 0x10,
+            TopRight    = 0x20,
+            BottomLeft  = 0x40,
+            BottomRight = 0x80,
+        };
+
+        // If the above scoped enums require operators then maybe they should be regular enums?
```
Comment (swichu91) [inline module-gui/gui/Common.hpp:123]: Let's resolve this issue instead of leaving comment ;)

File: module-gui/gui/widgets/Item.hpp [hunk 2]
```diff
@@ -79,7 +79,7 @@ namespace gui
         BoundingBox drawArea; // drawableArea would be more accurate
                               // maximal bounding box size
         /// gets bounding box for selected area
-        auto area(Area which = Area::Normal) -> BoundingBox &
```
Comment (SP2FET) [inline module-gui/gui/widgets/Item.hpp:82]: just asking, why did you change the trailing return to a standard one?

File: module-utils/utility/tests/unittest_utils.cpp [hunk 3]
```diff
@@ -526,6 +526,39 @@ TEST_CASE("Byte to hex")
...
+
+    SECTION("No")
+    {
+        REQUIRE((utils::startsWith("Abcde", "bcd") == false));
+    }
+
+    SECTION("Yes")
+    {
+        REQUIRE((utils::startsWith("Abcde", "Ab") == true));
+    }
```
Comment (SP2FET) [inline module-utils/utility/tests/unittest_utils.cpp:554]: I know it might be sufficient, but I think that it may be more verbose (I mean section names :D )

**Review by SP2FET — APPROVED** [review]

File: module-gui/gui/widgets/Item.hpp (hunk 2 above)
Comment (awulkiew) [reply module-gui/gui/widgets/Item.hpp:82]: Because:
1. For consistency, the standard return type is used for majority of functions in this class.
2. Trailing return type is not required in this case
3. The line is shorter

I don't have a strong opinion about it though. If you prefer the trailing type I can revert it.

But I think that if this is prefered then it should probably be done for all of the functions in the codebase or at least in this file. This way all of the function names would be aligned and the return type would always be at the end. So a reader would know where to look in all of the places. The function names would be easier to read but return types especially for functions with long list of parameters would be harder to read. Depends what we prefer.

File: module-gui/gui/Common.hpp (hunk 1 above)
Comment (awulkiew) [reply module-gui/gui/Common.hpp:123]: I mean this solution works and limits the scope of these operators. My point was that if we use these enums like integers performing bitwise operations and implicitly casting them to `bool` then maybe these shouldn't be class enums but regular enums instead? Since they are used as such.

Comment (awulkiew) [reply module-gui/gui/Common.hpp]: I removed the comment and one operator that was not used anywhere. I think this is still better than it was, i.e. the scope of the operators is limited, and can stay as it is for now.

File: module-utils/utility/tests/unittest_utils.cpp (hunk 3 above)
Comment (awulkiew) [reply module-utils/utility/tests/unittest_utils.cpp]: Done.
